### PR TITLE
feat: Refactor the cross chain contracts

### PR DESCRIPTION
**Pull request by fb-alexcq**

### CODE REVIEW

File: contracts/CrossChain/SgInfo.sol [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    mapping(uint16 => mapping(address => uint256)) public sgPoolIds; // chainId => token address => Stargate poolId
+
+    address public stargateRouter;
+    uint16 public sgChainIdLocal; // Stargate chainId on local chain
+
+    constructor(address _stargateRouter, uint16 _sgChainIdLocal) {
+        _initSgETHs();
+        _initSgPoolIds();
+        stargateRouter = _stargateRouter;
+        sgChainIdLocal = _sgChainIdLocal;
```
Comment (fb-alexcq) [inline contracts/CrossChain/SgInfo.sol:22]: Better to be moving this part to the beginning

File: contracts/CrossChain/SgInfo.sol [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        sgPoolIds[112][0x6626c47c00F1D87902fc13EECfaC3ed06D5E8D8a] = 20; // WOO
+        // Linea
+        sgPoolIds[183][0xe5D7C2a44FfDDf6b295A15c148167daaAf5Cf34f] = 13; // WETH
+        // Base
+        sgPoolIds[184][0xd9aAEc86B65D86f6A7B5B1b0c42FFA531710b6CA] = 1; // USDC
+        sgPoolIds[184][0x4200000000000000000000000000000000000006] = 13; // WETH
+    }
+
+    function setSgETH(uint16 chainId, address token) external onlyOwner {
+        require(token != address(0), "SgInfo: token cant be zero");
```
Comment (fb-alexcq) [inline contracts/CrossChain/SgInfo.sol:77]: Remove this line then.

File: contracts/CrossChain/SgInfo.sol [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    function setSgPoolId(
+        uint16 chainId,
+        address token,
+        uint256 poolId
+    ) external onlyOwner {
+        sgPoolIds[chainId][token] = poolId;
+    }
+
+    function setStargateRouter(address _stargateRouter) external onlyOwner {
+        require(_stargateRouter != address(0), "SgInfo: !_stargateRouter");
```
Comment (fb-alexcq) [inline contracts/CrossChain/SgInfo.sol:90]: Same here.

File: contracts/CrossChain/WooCrossChainRouterV4.sol [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+    constructor(
+        address _weth,
+        address _wooRouter,
+        address _sgInfo
+    ) {
+        weth = _weth;
+        wooRouter = IWooRouterV2(_wooRouter);
+        sgInfo = ISgInfo(_sgInfo);
+        sgChainIdLocal = sgInfo.sgChainIdLocal();
+        stargateRouter = IStargateRouter(sgInfo.stargateRouter());
```
Comment (fb-alexcq) [inline contracts/CrossChain/WooCrossChainRouterV4.sol:60]: These two variables are no longer needed.

File: contracts/CrossChain/SgInfo.sol (hunk 1 above)
Comment (Cphilo) [reply contracts/CrossChain/SgInfo.sol:20]: resolved

**Review by merlincdj — APPROVED** [review]

**Pull request merged**